### PR TITLE
Add import token

### PR DESCRIPTION
**Pull request by pilagod**
Add import token page, as the screen show shown below:

Screenshot 2024-07-23 at 6 06 23 PM

### CODE REVIEW

**Comment by oneleo** [comment]
This version is simpler as it won't fail, and the failure info is replaced by disabling the import token button.
I like the Symbol and Decimals preview, even though it's not in the Figma design.

Should we ask the designer to update the Figma design?

File: app/page/importToken/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      address.isEqual(t.address, tokenAddress)
+    )
+    if (!hasImported) {
+      await importToken(account.id, {
+        address: tokenAddress,
+        symbol: tokenSymbol,
+        decimals: Number(tokenDecimals),
+        balance: number.toHex(tokenBalance)
+      })
+    }
```
Comment (oneleo) [inline app/page/importToken/index.tsx:70]: Should we add `setToast("Token imported!", "success")` to the end of this condition according to the Figma design?
I tried, but got an `Invalid hook call` error...

File: app/page/importToken/index.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  const { provider } = useContext(ProviderContext)
+  const { setToast } = useContext(ToastContext)
+
+  const { importToken } = useAction()
+
+  const account = useAccount()
+  const tokens = useTokens()
+
+  const [tokenAddress, setTokenAddress] = useState("")
+  const [tokenAddressValid, setTokenAddressValid] = useState(false)
```
Comment (WIN-LIN) [inline app/page/importToken/index.tsx:27]: I think `tokenAddressValid` is a derived state variable, which can be calculated by token info (checking if there are symbol, token decimals)

File: app/page/importToken/index.tsx (hunk 1 above)
Comment (pilagod) [reply app/page/importToken/index.tsx]: Added in [eccf5cf](https://github.com/pilagod/waallet-extension/pull/178/commits/eccf5cf01598c31b516511f100471ba7e554f565)

File: app/page/importToken/index.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+  const [tokenFetching, setTokenFetching] = useState(false)
+
+  const onTokenAddressChanged = async (value: string) => {
+    setTokenAddress(value)
+
+    // Clear token information
+    setTokenSymbol("")
+    setTokenDecimals(0n)
+    setTokenBalance(0n)
```
Comment (WIN-LIN) [inline app/page/importToken/index.tsx]: how about we clear token information here if error occurs?

```
try {
 ....
} catch {
    // Clear token information
    setTokenSymbol("")
    setTokenDecimals(0n)
    setTokenBalance(0n)
 
}
```

Comment (pilagod) [reply app/page/importToken/index.tsx]: I wonder that if we clear them when error catched, when address is changed and waiting for fetching the new token info, the page would still present the previous token info.

File: app/page/importToken/index.tsx (hunk 2 above)
Comment (pilagod) [reply app/page/importToken/index.tsx]: Refined in [8b7ac9c](https://github.com/pilagod/waallet-extension/pull/178/commits/8b7ac9c811510bf92f74122023ffdcae49a7198f)

File: app/page/importToken/index.tsx (hunk 3 above)
Comment (WIN-LIN) [reply app/page/importToken/index.tsx]: You're right. We should clear token information when the user is attempting to input new address.

**Review by WIN-LIN — APPROVED** [review]
LGTM! Thanks for the work:)

Note: In the future, perhaps we can add debounce or cache to reduce the number of API calls.

**Pull request merged**